### PR TITLE
Implement downloadAttachment

### PR DESCRIPTION
**Pull request by bbartley**

### CODE REVIEW

File: sbol2/partshop.py [hunk 1]
```diff
@@ -344,6 +344,55 @@ def attachFile(self, top_level_uri, filepath):
...
+            filename = response.headers['Content-Disposition']
+            filename = filename[22:-1]  # remove extraneous text: attachment; filename="
+            if os.path.isdir(filepath):
+                filepath = posixpath.join(filepath, filename)
+            with open(filepath, 'wb') as filehandle:
+                filehandle.write(response.content)
+            return
+        if response.status_code == http.HTTPStatus.UNAUTHORIZED:
+            # HTTP 401
+            msg = 'You must login with valid credentials before attaching a file'
```
Comment (tcmitchell) [inline sbol2/partshop.py:384]: Copy paste error? Seems like the wrong error message.

File: test/test_partshop.py [hunk 2]
```diff
@@ -213,20 +213,63 @@ def test_uri2url_spoofed(self):
...
+        # attachment.source = CRISPR_LOCATION
+        # attachment.format = 'http://purl.org/NET/mediatypes/application/xml'
+        # md.attachments = [attachment.identity]
+        # sbh = sbol2.PartShop(TEST_RESOURCE)
+        # sbh.login(username, password)
+        # sbh.submit(doc, overwrite=1)
+        # attachment_uri = '{}/user/{}/{}/{}/{}'.format(sbh.getURL(), sbh.getUser(),
+        #                                               doc.displayId,
+        #                                               attachment.displayId,
+        #                                               attachment.version)
```
Comment (tcmitchell) [inline test/test_partshop.py:233]: Do we need all of this commented out code?

File: sbol2/partshop.py (hunk 1 above)
Comment (bbartley) [reply sbol2/partshop.py]: Changed to 'You must login with valid credentials before _downloading_ a file'

File: test/test_partshop.py (hunk 2 above)
Comment (bbartley) [reply test/test_partshop.py]: Cleaned up

**Pull request merged**